src/components/RegisterForm/RegisterForm.js

```diff
@@ -26,6 +38,58 @@ class RegisterForm extends Component {
     });
   }
 
+  onGetLocationSuccess(data) {
+    const fullAddress = data.results[0].formatted_address.split(', ');
+    const street = `${fullAddress[0]} ${fullAddress[1]}`;
+    const city = fullAddress[2];
+    const country = fullAddress[4];
+    const zipCode = fullAddress[5];
+    this.setState({
+      street,
+      city,
+      country,
+      zipCode
+    });
+  }
+
+  async getPosition(position) {
+    const lat = position.coords.latitude.toFixed(2);
+    const lon = position.coords.longitude.toFixed(2);
+    const proxy = 'https://cors-anywhere.herokuapp.com/';
+    const requestURL = `${proxy}https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lon}&key=${googleMapAPIKeyUserLocation}`;
+    const userAPI = new HttpService();
+    const { showMessage } = this.props;
+
+    try {
+      const response = await userAPI.get(requestURL);
+
+      this.onGetLocationSuccess(response.data);
+    } catch (error) {
+      showMessage(GET_LOCATION_FAILED);
+    }
+
+    setTimeout(() => {
+      this.setState({
+        isLoading: false
+      });
+    }, 500);
```

Comment:
Why are you using this set timeouts?

Reply:
To remove the preloader in five seconds after the request has been executed.

Reply:
I see it second time in PRs :)

Reply:
create a utility function then